lib/fluent/config/configure_proxy.rb

```diff
@@ -101,12 +116,19 @@ def merge_for_finalized(other)
         #  * overwrite param_name to escape duplicated name of instance variable
         #  * append params/defaults/sections which are missing in superclass
 
-        options = {
-          param_name: other.param_name,
-          required: (self.required.nil? ? other.required : self.required),
-          multi: (self.multi.nil? ? other.multi : self.multi),
-          final: true,
-        }
+        if other.final == false && overwrite?(other, :final)
+          raise ConfigError, "subclass cannot overwrite finalized base class's definition"
+        end
+
+        options = {}
+        options[:param_name] = other.param_name
+        options[:required] = @required.nil? ? other.required : self.required
+        options[:multi] = @multi.nil? ? other.multi : self.multi
+        options[:alias] = @alias.nil? ? other.alias : self.alias
+        options[:final]  = true
+
+        other.final = true
```

Comment:
I forgot to remove `other.final = true` in https://github.com/fluent/fluentd/pull/844/commits/5884b476593a2d6d47dba89a63913ec74670ebba